src/scripts/browserDeviceProfile.js

```diff
@@ -632,7 +650,7 @@ import browser from './browser';
         }
 
         if (canPlayHls() && options.enableHls !== false) {
-            if (hlsInFmp4VideoCodecs.length && hlsInFmp4VideoAudioCodecs.length && userSettings.preferFmp4HlsContainer() && (browser.safari || browser.tizen || browser.web0s)) {
+            if (hlsInFmp4VideoCodecs.length && hlsInFmp4VideoAudioCodecs.length && userSettings.preferFmp4HlsContainer() && canPlayHlsInFmp4()) {
```

Comment:
and `hlsInFmp4VideoCodecs` will be empty if it is not supported.
```suggestion
            if (hlsInFmp4VideoCodecs.length && hlsInFmp4VideoAudioCodecs.length && userSettings.preferFmp4HlsContainer()) {
```